test/Test/Toml/PrefixTree/Property.hs

```diff
@@ -90,7 +38,7 @@ prop_InsertLookup =  property $ do
 
 prop_InsertInsert :: Property
 prop_InsertInsert =  property $ do
-    t <- forAll genPrefixMap
+    t <- forAll $ genPrefixMap genVal
```

Comment:
Looks like you don't need to make `genPrefixMap` generator polymorphic since you call it only with `genVal` for now.